File: Python/modul7_kel46_stack.py
pilihan = 0
data = 0
count = 0
size = -1
stack = []
while count == 0:
    print("---Ini Stack---")
    print("1. PUSH Item")
    print("2. POP Item")
    print("3. Lihat Daftar Data")
    print("4. Lihat Data Teratas")
    pilihan = int(input("Masukkan Pilihan : "))
    count +=1
    if pilihan == 1:
        data = int(input("Data yang ditambahkan : "))
        stack.append(data)
        count = 0
        size+=1
        print ("")
    if pilihan == 2:
        if not stack :
            print("Data kosong")
            count = 0
            print ("")
        else :
            stack.pop()
            count = 0
            size-=1
            print ("")
    if pilihan == 3:
        print(stack)
        count = 0
        print ("")
    if pilihan == 4:
        if not stack :
            print("Data kosong")
            count = 0
            print ("")
        else :
            print(stack[size])
            count = 0
            print ("")

File: Java/modul7_kel46_stack.java
import java.util.Scanner;
import java.util.Stack;

@SuppressWarnings("unchecked")
public class modul7_kel46_stack {

    public static void main(String[] args) {
        int pilihan;
        int data;
        Stack result = new Stack();
        //looping until false condition
        do{
           //Displaying Menu
            System.out.println("---Ini Stack---");
            System.out.println("1. PUSH Item");
            System.out.println("2. POP Item");
            System.out.println("3. Lihat Daftar Data");
            System.out.println("4. Lihat Data Teratas");
            System.out.println("0. Keluar");
            Scanner input = new Scanner(System.in);
            System.out.print("Masukkan Pilihan : ");
            pilihan = input.nextInt();
            //condition for choice
            if(pilihan==1){
                System.out.print("Data yang ditambahkan : ");
                data = input.nextInt();
                result.push(data);
                System.out.println("");
            }
            else if(pilihan==2){
                if (result.isEmpty()) {
                    System.out.println("Data kosong");
                } else {
                    result.pop();
                    System.out.println("");
                }
            }
            else if(pilihan==3){
                System.out.print(result + " ");
                System.out.println("");
            }
            else if(pilihan==4){
                if (result.isEmpty()) {
                    System.out.println("Data kosong");
                } else {
                    System.out.println("Data teratas : "+result.peek());
                    System.out.println(""); 
                }
            }
            else if(pilihan==0){
                System.exit(0);
            }
            else{
                System.out.println("Pilihan Tidak Ada!!");
            }//end of condition
        } 
        while(pilihan!=0);//end looping
    }
}